src/main/java/hudson/plugins/gradle/injection/MavenInjectionEnvironmentContributor.java

```diff
@@ -0,0 +1,105 @@
+
+@Extension
+public class MavenInjectionEnvironmentContributor extends EnvironmentContributor implements MavenInjectionAware {
+
+    private static final MavenOptsHandler MAVEN_OPTS_HANDLER = new MavenOptsHandler(
+            MAVEN_EXT_CLASS_PATH_PROPERTY_KEY,
+            BUILD_SCAN_UPLOAD_IN_BACKGROUND_PROPERTY_KEY,
+            GRADLE_ENTERPRISE_ALLOW_UNTRUSTED_SERVER_PROPERTY_KEY,
+            GRADLE_ENTERPRISE_URL_PROPERTY_KEY
+    );
+
+    private final MavenExtensionsHandler extensionsHandler = new MavenExtensionsHandler();
+
+    @Override
+    public void buildEnvironmentFor(@Nonnull Run run, @Nonnull EnvVars envs, @Nonnull TaskListener listener) {
+        Executor executor = run.getExecutor();
+        if (executor == null) {
+            return;
+        }
+
+        Node node = executor.getOwner().getNode();
+        if (node == null) {
+            return;
+        }
+
+        FilePath nodeRootPath = node.getRootPath();
+        if (nodeRootPath == null) {
+            return;
+        }
+
+        if (!isInjectionEnabled(node)) {
+            return;
+        }
+
+        InjectionConfig config = InjectionConfig.get();
+        String server = config.getServer();
+
+        List<FilePath> extensions = new LinkedList<>();
+        extensions.add(extensionsHandler.getExtensionLocation(MavenExtensionsHandler.MavenExtension.GRADLE_ENTERPRISE, nodeRootPath));
+        if (config.isInjectCcudExtension()) {
+            extensions.add(extensionsHandler.getExtensionLocation(MavenExtensionsHandler.MavenExtension.CCUD, nodeRootPath));
+        }
+
+        boolean isUnix = isUnix(node);
+
+        List<SystemProperty> systemProperties = new ArrayList<>();
+        systemProperties.add(new SystemProperty(MAVEN_EXT_CLASS_PATH_PROPERTY_KEY, constructExtClasspath(extensions, isUnix)));
+        systemProperties.add(new SystemProperty(BUILD_SCAN_UPLOAD_IN_BACKGROUND_PROPERTY_KEY, "false"));
+
+        systemProperties.add(new SystemProperty(GRADLE_ENTERPRISE_URL_PROPERTY_KEY, server));
+        if (config.isAllowUntrusted()) {
+            systemProperties.add(new SystemProperty(GRADLE_ENTERPRISE_ALLOW_UNTRUSTED_SERVER_PROPERTY_KEY, "true"));
+        }
+
+        envs.put(MAVEN_OPTS, MAVEN_OPTS_HANDLER.create(envs, systemProperties));
+
+        // Configuration needed to support https://plugins.jenkins.io/maven-plugin/
+        extensions.add(extensionsHandler.getExtensionLocation(MavenExtensionsHandler.MavenExtension.CONFIGURATION, nodeRootPath));
+
+        envs.put(JENKINSGRADLEPLUGIN_MAVEN_PLUGIN_CONFIG_EXT_CLASSPATH, constructExtClasspath(extensions, isUnix));
+        envs.put(JENKINSGRADLEPLUGIN_MAVEN_PLUGIN_CONFIG_SERVER_URL, server);
+        if (config.isAllowUntrusted()) {
+            envs.put(JENKINSGRADLEPLUGIN_MAVEN_PLUGIN_CONFIG_ALLOW_UNTRUSTED_SERVER, "true");
+        } else {
+            envs.remove(JENKINSGRADLEPLUGIN_MAVEN_PLUGIN_CONFIG_ALLOW_UNTRUSTED_SERVER);
```

Comment:
Is it for backward compatibility? Let's add a comment.

Reply:
This is actually not needed, as if the `config.isAllowUntrusted` is true, we will set it here as well (which won't clash with the env var on the node). And if it is disabled, that means that we don't need to do anything here, since it was already like that or it was turned off in which case our cleanup part in `MavenBuildScanInjection` deletes it on configuration save.